lib/src/logic/actions/move_tile_action.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:link_five/src/logic/actions/action_helpers.dart';
+import 'package:link_five/src/logic/game_action.dart';
+import 'package:link_five/src/model/game/game_state.dart';
+import 'package:link_five/src/model/game/player_color.dart';
+import 'package:link_five/src/model/game/tile.dart';
+import 'package:link_five/src/model/game/tile_location.dart';
+
+class MoveTileAction extends GameAction {
+  TileLocation source;
+  TileLocation destination;
+
+  MoveTileAction({
+    required PlayerColor playerColor,
+    required this.source,
+    required this.destination,
+  }) : super(playerColor: playerColor);
+
+  @override
+  bool isPermitted(GameState gameState) {
+    final isGameBoardEmpty = gameState.tiles.isEmpty;
```

Comment:
No need to check if the game board is empty since you can't move tiles on an empty game board.